specs/core/0_beacon-chain.md

```diff
@@ -988,33 +1002,31 @@ def get_randao_mix(state: BeaconState,
 
 ```python
 def get_active_index_root(state: BeaconState,
-                          slot: int) -> Bytes32:
+                          epoch: int) -> Bytes32:
     """
     Returns the index root at a recent ``slot``.
     """
-    state_epoch = state.slot // EPOCH_LENGTH
-    given_epoch = slot // EPOCH_LENGTH 
-    assert state_epoch < given_epoch + LATEST_INDEX_ROOTS_LENGTH
-    assert given_epoch <= state_epoch
-    return state.latest_index_roots[given_epoch % LATEST_INDEX_ROOTS_LENGTH]
+    assert current_epoch(state) < epoch + LATEST_INDEX_ROOTS_LENGTH
+    assert epoch <= current_epoch(state)
+    return state.latest_index_roots[epoch % LATEST_INDEX_ROOTS_LENGTH]
 ```
 
 #### `generate_seed`
 
 ```python
 def generate_seed(state: BeaconState,
-                  slot: int) -> Bytes32:
+                  epoch: int) -> Bytes32:
     """
-    Generate a seed for the given ``slot``.
+    Generate a seed for the given ``epoch``.
     """
-    if slot < SEED_LOOKAHEAD:
-        randao_mix_slot = GENESIS_SLOT
+    if epoch < SEED_LOOKAHEAD:
```

Comment:
Wait, why are we checking `slot` or `epoch` against a constant anyway? Shouldn't this be `epoch < current_epoch(state) + SEED_LOOKAHEAD`? Also, shouldn't `SEED_LOOKAHEAD` be denominated in epochs?

Reply:
This is preventing underflow on the `randao_mix_epoch` when current_epoch is less than `SEED_LOOKAHEAD`.

Reply:
Aah I see.

Reply:
changed this to remove the special case.

```python
randao_mix_epoch = epoch + LATEST_RANDAO_MIXES_LENGTH - SEED_LOOKAHEAD
```